pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -184,11 +187,102 @@ data:
 ---
 {{- end}}
 `
+
+	// vmDeploymentYaml aims to simulate a VM, but instead of managing the complex test setup of spinning up a VM,
+	// connecting, etc we run it inside a pod. The pod has pretty much all Kubernetes features disabled (DNS and SA token mount)
+	// such that we can adequately simulate a VM and DIY the bootstrapping.
+	vmDeploymentYaml = `
+{{- $cluster := .Cluster }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ $.Service }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      istio.io/test-vm: {{ $.Service }}
+  template:
+    metadata:
+      annotations:
+        # Sidecar is inside the pod to simulate VMs - do not inject
+        sidecar.istio.io/inject: "false"
+      labels:
+        # Label should not be selected. We will create a workload entry instead
+        istio.io/test-vm: {{ $.Service }}
+    spec:
+      # Disable kube-dns, to mirror VM
+      dnsPolicy: Default
+      # Disable service account mount, to mirror VM
+      automountServiceAccountToken: false
+      containers:
+      - name: istio-proxy
+        image: {{ $.Hub }}/app_sidecar:{{ $.Tag }}
+        #image: {{ $.Hub }}/app_sidecar:{{ $.Tag }}
+        imagePullPolicy: {{ $.PullPolicy }}
+        securityContext:
+          capabilities:
+            add:
+            - NET_ADMIN
+          runAsUser: 1338
+          runAsGroup: 1338
+        command:
+        - bash
+        - -c
+        - |-
+          sudo sh -c 'echo ISTIO_SERVICE_CIDR=* > /var/lib/istio/envoy/cluster.env'
+          sudo sh -c 'echo ISTIO_PILOT_PORT={{$.VM.IstiodPort}} >> /var/lib/istio/envoy/cluster.env'
+          sudo sh -c 'echo "{{$.VM.IstiodIP}} istiod.istio-system.svc" >> /etc/hosts'
+          sudo sh -c 'echo "1.1.1.1 pod.{{$.Namespace}}.svc.cluster.local" >> /etc/hosts'
+
+          # TODO: run with systemctl?
+          sudo -E /usr/local/bin/istio-start.sh&
+          /usr/local/bin/server --cluster "{{ $cluster }}" \
+{{- range $i, $p := $.ContainerPorts }}
+{{- if eq .Protocol "GRPC" }}
+             --grpc \
+{{- else if eq .Protocol "TCP" }}
+             --tcp \
+{{- else }}
+             --port \
+{{- end }}
+             "{{ $p.Port }}" \
+{{- end }}
+        env:
+        # We use token not certs
+        - name: PROV_CERT
+          value: ""
+        # By default we do not capture inbound. For these tests we will mark as capture all
+        - name: ISTIO_INBOUND_PORTS
+          value: "*"
```

Comment:
it would be better to run the VM tests without any sidecar capture and stick to using the Sidecar crd. Otherwise, we will have no definitive way of knowing if the VM things worked properly or not since we are emulating everything inside pods

Reply:
See comment above - we should test both in my opinion